amazon-link-engine/trunk/amazon-link-engine.php

```diff
@@ -249,6 +255,17 @@ function genius_admin_notice(){
   }
 }
 
+//Gets the state of the checkbox for onClick functionality of link conversion
+
+function ale_get_on_click_checkbox_state() {
+	$ale_on_click_checkbox_state = get_option('genius_ale_urls_on_click');
+
+	if ($ale_on_click_checkbox_state == 'yes') return true;
```

Comment:
this should be just return $ale_on_click_checkbox_state == 'yes'  

That already returns boolean values